proof_ipa.go

```diff
@@ -524,20 +577,20 @@ func PostStateTreeFromStateDiff(preroot VerkleNode, statediff StateDiff) (Verkle
 
 	for _, stemstatediff := range statediff {
 		var (
-			values     = make([][]byte, NodeWidth)
-			overwrites bool
+			values    = make([][]byte, NodeWidth)
+			overwrite bool
 		)
 
-		for _, suffixdiff := range stemstatediff.SuffixDiffs {
-			if /* len(suffixdiff.NewValue) > 0 - this only works for a slice */ suffixdiff.NewValue != nil {
```

Comment:
This previous `if` (that was kept in Type 2) was removed. This can also be removed in Type 2 as I mentioned in my review in that PR.

Both in `Inserts` and `Updates`, we can't have "nil" or empty values in `New`.